torch/csrc/jit/tensorexpr/kernel.h

```diff
@@ -211,6 +211,8 @@ class TensorExprKernel {
   std::unordered_map<int64_t, Tensor*> tensors_;
   std::unordered_map<int64_t, VarHandle> scalars_;
   std::unique_ptr<CodeGen> codegen_;
+  size_t current_hashcode = 0;
```

Comment:
I don't think `current_hashcode` actually needs to be part of the TensorExprKernel state.  We already have a hash->CodeGen map.  I bet you can just hash the at::Device, look that up in the map and decide what to do from there.  You can probably even get rid of `backend_type_` below.

Reply:
thanks, I am getting rid of current_hashcode, and hashcode.
I'll get rid or backend_type_ in a followup diff.